### PR TITLE
llvm-amdgpu: conflict with macOS

### PR DESCRIPTION
**Pull request by stephenswat**
Currently, the llvm-amdgpu package doesn't compile on MacOS, but it is also not marked as a conflict. This causes problems because it seems that Spack is very happy to pull in llvm-amdgpu as the default package to satisfy any virtual libllvm dependency, which can cause dependent specs to fail to install on MacOS. This commit marks a conflict between this llvm package and the Darwin platform.

### CODE REVIEW

**Review by alecbcs — APPROVED** [review]
Looks good to me. Waiting on final approval from @renjithravindrankannath @srekolam or @haampie. 
Thanks @stephenswat!

**Review by srekolam — APPROVED** [review]
LGTM .thanks for the changes.

**Review by renjithravindrankannath — APPROVED** [review]
LGTM

**Comment by alecbcs** [comment]
@spackbot run pipeline

**Comment by stephenswat** [comment]
@spackbot run pipeline

**Comment by stephenswat** [comment]
It appears the E4S CI job is failing due to a timeout in an HTTP request. Does anyone recognise this issue?

**Comment by alecbcs** [comment]
Yes 😢 

@jjnesbitt @mvandenburgh is the suggestion to just keep restarting the job until it it gets through the network?

**Comment by stephenswat** [comment]
@spackbot run pipeline

**Comment by stephenswat** [comment]
Thanks! I'll keep trying.

**Comment by stephenswat** [comment]
@spackbot run pipeline

**Comment by mvandenburgh** [comment]
> Yes 😢
> 
> @jjnesbitt @mvandenburgh is the suggestion to just keep restarting the job until it it gets through the network?

Unfortunately I think that's the only recourse here :disappointed: .

This particular issue with `vtk-m` is actually not related to the cluster networking issues, though. It's happening because the request to gitlab.kitware.com is failing, which is being looked into

**Pull request merged**
